### PR TITLE
Added a loss function name and loss function params option to the pytorch models

### LINKED ISSUE #144: Recording loss functions in run.json
If we make any changes to loss functions, we will lose reproducibility of run.json files.

### PR DESCRIPTION
**Pull request by erikhuck**
closes #144

### CODE REVIEW

**Comment by bjschoenfeld** [comment]
I don't see where the `loss_function_params` get passed into the loss function. It would be good to add a loss function that has params to test this code.

**Comment by orionw** [comment]
I've added a `categorical` option to `tuningdeap` now.  When this is approved I can test it works on DNA and then update `tuningdeap` on pypi so that we can hyperparameter tune it genetically if we want.
